test/over_react_redux/fixtures/counter.dart

```diff
@@ -48,35 +48,35 @@ class CounterComponent extends UiComponent2<CounterProps> {
         ..addTestId('button-increment')
         ..onClick = (_) {
           if (props.increment != null) {
-            props.increment();
+            props.increment!();
           } else if (props.dispatch != null) {
-            props.dispatch(IncrementAction());
+            props.dispatch!(IncrementAction());
```

Comment:
I think I may have overlooked this in my PR... should `props.dispatch` actually be nullable? Feels wrong / disjointed when compared to `useDispatch` - which does not have a nullable return signature.

Reply:
Hmm that's a good question... On one hand, yeah, making it non-nullable seems like it'd be correct. On the other hand, that would also make it required, and end up throwing/linting about `dispatch` being missing when consuming a connect component. 

Hmmm.

Perhaps we could come up with a way to indicate a prop is non-nullable at render time, but doesn't need to be specified by the consumer? Like, via an annotation or something. 

I wonder if this case is common enough that we'll actually need it... I bet we'll run into this when dealing with components with props that are always cloned onto them by parent components, but don't need to or shouldn't be specified by the consumer.